include/dxc/DxilContainer/DxilRuntimeReflection.h

```diff
@@ -238,32 +238,20 @@ template<typename _T>
 class RecordTraits {
 public:
   static constexpr const char *TypeName() {
-#ifdef _WIN32
-    static_assert(false, "");
```

Comment:
Do we have a way to do this at compile time?

Reply:
This pattern doesn't work at compile time because a non-dependent static assert is defined to be evaluated at parse, not instantiation. MSVC incorrectly implements the C++ standard.

Looking at what this is doing, you can convert this to a linker error by deleting the method body entirely, which is probably better than what I did here.

Reply:
Update pushed.